src/main/webapp/consentGroupContainer/ConsentGroupContainer.js

```diff
@@ -0,0 +1,191 @@
+
+export const ConsentGroupContainer = hh(class ConsentGroupContainer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      currentStepIndex: 0,
+      historyContent: '',
+      commentsContent: '',
+      dialogContent: '',
+      defaultActive: 'review'
+    };
+  }
+
+  componentDidMount() {
+    this.getHistory();
+    this.getComments();
+  }
+
+  updateDetailsStatus = (status) => {
+    this.getHistory();
+    this.props.updateDetailsStatus(status);
+  };
+
+  updateDocumentsStatus = (status) => {
+    this.getHistory();
+    this.props.updateDocumentsStatus(status);
+  };
+
+  updateAdminOnlyStatus = (status) => {
+    this.getHistory();
+    this.props.updateAdminOnlyStatus(status);
+  };
+
+  updateContent = () => {
+    this.getComments();
+    this.getHistory();
+  };
+
+  // comments
+  getComments() {
+    ProjectMigration.getComments(this.props.serverURL, this.props.consentKey).then(resp => {
+      this.setState(prev => {
+        prev.commentsContent = resp.data;
+        return prev;
+      }, () => {
+        this.initializeComments();
+      });
+    })
+  };
+
+  initializeComments() {
+    tinymce.remove();
+    $.fn.dataTable.moment('MM/DD/YYYY hh:mm:ss');
+    if (!$.fn.dataTable.isDataTable("#comments-table")) {
+      $("#comments-table").DataTable({
+        dom: '<"H"Tfr><"pull-right"B><div>t</div><"F"lp>',
+        buttons: ['excelHtml5', 'csvHtml5', 'print'],
+        language: { search: 'Filter:' },
+        pagingType: "full_numbers",
+        order: [1, "desc"]
+      });
+    }
+    this.initializeEditor();
+  }
+
+  initializeEditor() {
+    tinymce.init({
+      selector: 'textarea.editor',
+      width: '100%',
+      menubar: false,
+      statusbar: false,
+      plugins: "paste",
+      paste_data_images: false
+    });
+  }
+
+  // history
+  getHistory() {
+    ProjectMigration.getHistory(this.props.serverURL, this.props.consentKey).then(resp => {
+      this.setState(prev => {
+        prev.historyContent = resp.data;
+        return prev;
+      }, () => {
+        this.initializeHistory();
+      });
+    })
+  };
+
+  initializeHistory() {
+    $.fn.dataTable.moment('MM/DD/YYYY hh:mm:ss');
+    if (!$.fn.dataTable.isDataTable("#history-table")) {
+      $("#history-table").DataTable({
+        dom: '<"H"Tfr><"pull-right"B><div>t</div><"F"lp>',
+        buttons: ['excelHtml5', 'csvHtml5', 'print'],
+        language: { search: 'Filter:' },
+        pagingType: "full_numbers",
+        order: [1, "desc"]
+      });
+    }
+  }
+
+  render() {
+    return (
+      div({ className: "headerBoxContainer" }, [
+        div({ className: "containerBox" }, [
+          MultiTab({ defaultActive: this.props.tab === "" ? this.state.defaultActive : this.props.tab },
+            [
+              div({
+                key: "review",
+                title: "Cohort Details",
+              }, [
+                  h(ConsentGroupReview, {
```

Comment:
I question whether these components should take all of these constant values as props or if they should be getting them from a utility like the ajax.js lib. There is a lot of copy/paste that I think can be abstracted away from component creation and instead accessed from the component directly.